lib/system.nim

```diff
@@ -2477,7 +2477,7 @@ proc `$`*[T: tuple|object](x: T): string =
   ## of `x`. Example:
   ##
   ## .. code-block:: nim
-  ##   $(23, 45) == "(23, 45)"
+  ##   $(23, 45) == "(Field0: 23, Field1: 45)"
```

Comment:
Wouldn't the proper fix be something like:

```Nim
func `$`(x: tuple): string =
  result = "("
  for idx, val in fieldPairs(x):
    if idx != 0:
      result.add ", "
    result.add $val
  result.add ')'
```

(untested)

Reply:
This would lose information for named tuples.

Reply:
IMO the optional names in tuples are aliases to ease writing and reading code, the tuple model however remains as an ordered list of N values. So for creating a string representation I would ignore the names and just dump the values. Another thing would be for debugging (`repr`), in which case it should output the field names.

Reply:
@drslump please, consider ignoring them only when they start with Field0: this way you'll have a nice repr for both kinds of tuples 98% of the time

Reply:
My thinking is that `$` means **to string**, it should not leak implementation details.

Has it been considered to introduce a `$$` operator convention to generate a print-debug oriented *to-string*?

Reply:
@drslump nope there is no ``$$`` operator. The ``$`` operator is already for print-debug.

Reply:
``marshal.nim`` has a ``$$`` operator. :-)

Reply:
> please, consider ignoring them only when they start with Field0

`isNamedTuple` (see my suggested implementation here https://github.com/nim-lang/Nim/pull/10010#issuecomment-448013600) is more robust than checking if they start with `Field0` as it relies on `compiles(t.Field0)`, which would be false for unnamed tuple even if it has a field "Field0" (and also more efficient, since only 1st field needs to be checked)

Reply:
=> https://github.com/nim-lang/Nim/pull/10070